### PR TITLE
Fix image tagging for ACR Build scenario in 'containerapp up'

### PR DESCRIPTION
**Pull request by cormacpayne**
Fix image tagging for ACR Build scenario in `containerapp up`

A bug was introduced in PR https://github.com/Azure/azure-cli-extensions/pull/6283 that prevented the image tag created for the ACR Build scenario from being propagated globally via `self.image` to the Container App update call, meaning the update had no reference to the DateTime tag created and used the `latest` tag, which doesn't exist (for images created by this flow).

This PR aims to fix this bug by ensuring that the ACR Build and ACR Task flows will use the same tag created and set globally, and the Buildpack flow will use a separate variable to store the tag it creates and sets globally.

The three flows have been tested and verified locally.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

### Related command

`az containerapp up`

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

~- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).~

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Thank you for your contribution! We will review the pull request and get back to you soon.

**Review by daniv-msft — APPROVED** [review]

**Review by daniv-msft — APPROVED** [review]

**Review by harryli0108 — APPROVED** [review]
LGTM with one quick question. I think this test achieved a good test coverage with the general use scenario here (both acr build w dockerfile as well as buildpacks build w/o dockerfile), but do you plan to extend the test to cover error handling and/or passing wrong info as argument input?

**Comment by zhoxing-ms** [comment]
Could you please move `source_using_dockerfile`, `source_using_acr_task`, and `source_using_buildpack` uniformly to folder `src/containerapp/azext_containerapp/tests/latest/data`? This facilitates unified maintenance and management of these test dependent files in the future

**Comment by zhoxing-ms** [comment]
Please add the description of the PR change into the `HISTORY.rst`. If you want to release a new extension for this PR, please also upgrade the extension version defined in the `setup.py`

**Comment by cormacpayne** [comment]
@zhoxing-ms -- thanks for the comments! I just resolved them.

@harryli0108 -- we can look into more robust testing in the future, right now we're focused on adding tests to catch regressions in existing scenarios.

File: src/containerapp/azext_containerapp/_up_utils.py [hunk 1]
```diff
@@ -575,11 +576,12 @@ def run_acr_build(self, dockerfile, source, quiet=False, build_from_source=False
...
                 logger.warning("Attempting to build image using buildpacks...")
+                buildpack_image_name_with_tag = image_name_with_tag
                 run_image_tag = get_latest_buildpack_run_tag("aspnet", "7.0")
                 if run_image_tag is not None:
-                    image_name = f"{image_name}:{run_image_tag}-{tag_now_suffix}"
-                self.build_container_from_source_with_buildpack(image_name, source)
-                self.image = self.registry_server + "/" + image_name
+                    buildpack_image_name_with_tag = f"{image_name}:{run_image_tag}-{tag_now_suffix}"
+                self.build_container_from_source_with_buildpack(buildpack_image_name_with_tag, source)
+                self.image = self.registry_server + "/" + buildpack_image_name_with_tag
```
Comment (Juliehzl) [inline src/containerapp/azext_containerapp/_up_utils.py:584]: check indent. Should self.build_container_from_source_with_buildpack and self.image in the same part?

**Review by Juliehzl — APPROVED** [review]

File: src/containerapp/azext_containerapp/_up_utils.py (hunk 1 above)
Comment (daniv-msft) [reply src/containerapp/azext_containerapp/_up_utils.py:584]: Thanks for your comment! As discussed on Teams, we confirm the indent looks good.

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**